### PR TITLE
Refactoring DSCNodeConfigurationCreate Parameters

### PR DESCRIPTION
**Pull request by vrdmr**
Refactoring DSCNodeConfiguration to prevent the Autorest to not merge the models while generating the dot net client.
Also, fixed the DSCNodeConfiguration (2018-01-15) Get call to correct the response code. The service returns the 201 and fixing this to currently unblock out azure powershell release.

----
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by ravbhatnagar** [comment]
@vrdmr  - Looks like the shape of a model has changed. This is a existing APi-version. Do you intend to make this change for an existing api-version?

File: specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/definitions.json [hunk 1]
```diff
@@ -1482,32 +1482,6 @@
       },
       "description": "Definition of the DSC Meta Configuration."
     },
-    "DscNodeConfigurationCreateOrUpdateParameters": {
```
Comment (vrdmr) [inline specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/definitions.json]: Moved these definitions to 2015-10-31/DscNodeConfiguration.json

File: specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/dscNodeConfiguration.json [hunk 2]
```diff
@@ -293,6 +293,78 @@
       ],
       "description": "Definition of the dsc node configuration."
     },
+    "ContentSource": {
```
Comment (vrdmr) [inline specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/dscNodeConfiguration.json]: Added the definitions from Defnitions.json to this file.

File: specification/automation/resource-manager/Microsoft.Automation/stable/2018-01-15/dscNodeConfiguration.json [hunk 3]
```diff
@@ -438,10 +439,6 @@
           "$ref": "#/definitions/ContentSource",
           "description": "Gets or sets the source."
         },
-        "name": {
```
Comment (vrdmr) [inline specification/automation/resource-manager/Microsoft.Automation/stable/2018-01-15/dscNodeConfiguration.json]: Corrected the model. The Name is not in the top-level parameter.

File: specification/automation/resource-manager/Microsoft.Automation/stable/2018-01-15/examples/createOrUpdateDscNodeConfiguration.json [hunk 4]
```diff
@@ -3,7 +3,6 @@
         "subscriptionId": "subid",
         "resourceGroupName": "rg",
         "automationAccountName": "myAutomationAccount20",
-        "credentialName": "myCredential",
```
Comment (vrdmr) [inline specification/automation/resource-manager/Microsoft.Automation/stable/2018-01-15/examples/createOrUpdateDscNodeConfiguration.json]: This was wrong param being sent in the example.

**Comment by vrdmr** [comment]
@ravbhatnagar - The model change is to make it at par with the service now. The "name" was added. (I've added comments for each file now.

**Comment by vrdmr** [comment]
@sergey-shandar - The 2015-refactoring is just moving the definitions around to prevent the autorest generation issues that we are seeing in the SDK due to the new 2018-01-15 API spec for DSCNodeConfiguration.

**Comment by vrdmr** [comment]
@ravbhatnagar @sergey-shandar Could you please take a look at this PR. Thanks.

**Comment by ravbhatnagar** [comment]
@vrdmr - Sounds good. Since the service never used to return this model, it should be fine. Signing off from ARM side.

**Comment by vrdmr** [comment]
Thanks, @ravbhatnagar. @sergey-shandar - Could you please take a look at this PR.

**Review by sergey-shandar — APPROVED** [review]

**Comment by vrdmr** [comment]
Thanks, @sergey-shandar.

**Pull request merged**